### ISSUE TITLE
[BUG] Feature photos for people are changing after updating to 1.90.0

### ISSUE THREAD

**Issue by eygraber** [immich-app/immich#5540]
### The bug

The feature photos that I set for people are no longer there. Instead there is a different photo from that person being used. It's not for all people, and it didn't seem to happen right after updating, but only after using the website for a little bit.

### The OS that Immich Server is running on

Ubuntu 22.04

### Version of Immich Server

v1.90.0

### Version of Immich Mobile App

v1.90.0

### Platform with the issue

- [X] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
version: "3.8"

services:
  immich-server:
    user: 1000:1001
    container_name: immich_server
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    command: [ "start.sh", "immich" ]
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
    env_file:
      - .env
    ports:
      - "2283:3001"
    depends_on:
      - redis
      - database
      - typesense
    restart: always

  immich-microservices:
    user: 1000:1001
    container_name: immich_microservices
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    # extends:
    #   file: hwaccel.yml
    #   service: hwaccel
    command: [ "start.sh", "microservices" ]
    volumes:
      - /disks/storage/immich/volumes/geocoding:/usr/src/app/.reverse-geocoding-dump
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
    env_file:
      - .env
    depends_on:
      - redis
      - database
      - typesense
    restart: always

  immich-machine-learning:
    user: 1000:1001
    container_name: immich_machine_learning
    image: ghcr.io/immich-app/immich-machine-learning:${IMMICH_VERSION:-release}
    volumes:
      - /disks/storage/immich/volumes/model-cache:/cache
    env_file:
      - .env
    restart: always

  typesense:
    container_name: immich_typesense
    image: typesense/typesense:0.24.1@sha256:9bcff2b829f12074426ca044b56160ca9d777a0c488303469143dd9f8259d4dd
    environment:
      - TYPESENSE_API_KEY=${TYPESENSE_API_KEY}
      - TYPESENSE_DATA_DIR=/data
      # remove this to get debug messages
      - GLOG_minloglevel=1
    volumes:
      - /disks/storage/immich/volumes/tsdata:/data
    restart: always

  redis:
    container_name: immich_redis
    image: redis:6.2-alpine@sha256:70a7a5b641117670beae0d80658430853896b5ef269ccf00d1827427e3263fa3
    restart: always

  database:
    user: 1000:1001
    container_name: immich_postgres
    image: postgres:14-alpine@sha256:28407a9961e76f2d285dc6991e8e48893503cc3836a4755bbc2d40bcc272a441
    env_file:
      - .env
    environment:
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_DB: ${DB_DATABASE_NAME}
    volumes:
      - /disks/storage/immich/volumes/pgdata:/var/lib/postgresql/data
    restart: always
```

### Your .env content

```Shell
# You can find documentation for all the supported env variables at https://immich.app/docs/install/environment-variables

PUID=1000
GUID=1001

# The location where your uploaded files are stored
UPLOAD_LOCATION=/disks/storage/immich/library

# The Immich version to use. You can pin this to a specific version like "v1.71.0"
IMMICH_VERSION=release

# Connection secrets for postgres and typesense. You should change these to random passwords
TYPESENSE_API_KEY=...
DB_PASSWORD=...

TZ=America/New_York

# The values below this line do not need to be changed
###################################################################################
DB_HOSTNAME=immich_postgres
DB_USERNAME=postgres
DB_DATABASE_NAME=immich

REDIS_HOSTNAME=immich_redis
```

### Reproduction steps

```bash
Not fully sure, just noticed that the feature photo changed for some people.
```

**Comment by martabal** [comment]
If you re-assigned the face which is the feature photo of the edited person, yes it's supposed to change the feature photo. You don't want to people with the same feature photo

**Comment by eygraber** [comment]
I'm not sure I understand. This is what happened

1. Person A was tagged in a photo they were not in
2. I reassigned the face in that photo to Person B
3. Person B's feature photo changed

**Comment by martabal** [comment]
It means the Person B `faceAssetId` was set to null :thinking: When was that person created ? If it was a long time ago, I think one release mentioned to re-generate the feature photos to avoid this issue.

**Comment by eygraber** [comment]
Person B was probably created a couple of months ago. I searched the release notes for "feature photo" and only found two mentions; when the ability to change them was added and something related to websockets. I'd appreciate any more info if someone has :pray:

**Comment by raisinbear** [comment]
Yeah I noticed the same with following scenario: person wrongfully identified as person „A“, when reassigned to person „B“, the feature photo for „B“ changes to a presumably random one. This process did not alter the feature photo for person „A“ and was not the feature photo for person „A“. Could reproduce this multiple times. Hope this helps!
Awesome feature by the way! Thanks so much for the effort @martabal ! 🤩

**Comment by eygraber** [comment]
@martabal sorry if I misunderstood, but is there anything we can do to prevent this from happening?

**Comment by jrasm91** [comment]
We used to set a feature photo and not track what asset it came from. With the recent changes a feature photo _has_ to have a reference back to the asset it came from, otherwise we'll lose it and pick a random one. This is just a problem with how the implementation was originally built and it should not happen on a regular basis.

**Comment by eygraber** [comment]
Anything I can do to prevent it from happening further, or I just have to deal with it as it happens?

**Comment by martabal** [comment]
You can set the feature photo with the "Change feature" option on the person page. But there's no easy way to know which person has their `faceAssetId` set to null, you'll have to check in the database yourself

**Comment by eygraber** [comment]
Got it thanks! Seems like this could be closed then?

**Comment by eygraber** [comment]
This is still happening on v1.91.4 for people that shouldn't have `faceAssetId == null` (i.e. I've already set their feature photo recently).

It seems to happen after reassigning a face to that person; is this something that https://github.com/immich-app/immich/pull/5796 addresses, or is there something else going on?

**Comment by martabal** [comment]
This has nothing to do with this.

Can you make sure you don't have people with faceAssetId set to null with `SELECT * FROM "person" WHERE "faceAssetId" IS NULL`

**Comment by eygraber** [comment]
I checked and I have 86 with `faceAssetId != null` and 255 with `faceAssetId == null` (this is only for people that have a name; there's 12k more that don't have a name).

For the ones that are null, I've definitely set a feature photo for them recently.

**Comment by eygraber** [comment]
Also finding that some newly created people have a null `faceAssetId`. Can this be re-opened, or should I file a new issue?
